test-suite/shared/src/test/require-jdk8/org/scalajs/testsuite/javalib/lang/LongTestOnJDK8.scala

```diff
@@ -0,0 +1,620 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js Test Suite        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2015, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+package org.scalajs.testsuite.javalib.lang
+
+import org.junit.Test
+import org.junit.Assert._
+
+import org.scalajs.testsuite.utils.AssertThrows._
+
+import java.lang.{Long => JLong}
+
+class LongTestOnJDK8 {
+
+  final val MinRadix = Character.MIN_RADIX
+  final val MaxRadix = Character.MAX_RADIX
+
+  // Explicitly define these as `var`'s to avoid any compile-time constant folding
+  val MaxValue: Long = Long.MaxValue
+  val MinValue: Long = Long.MinValue
+
+  @Test def parseUnsignedLong(): Unit = {
+    def test(s: String, v: Long, radix: Int = 10): Unit = {
+      if (JLong.parseUnsignedLong(s, radix) != v)
+        println(s"$s, $v, $radix")
+      assertTrue(JLong.parseUnsignedLong(s, radix) == v)
+      if (radix == 10)
+        assertTrue(JLong.parseUnsignedLong(s) == v)
```

Comment:
Use `assertEquals`. With JUnit it will do the right thing (with Jasmine it wouldn't for Longs).